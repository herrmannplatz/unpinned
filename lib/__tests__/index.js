/* eslint-env jest */
const unpinned = require('../')
const { resolve } = require('path')

describe('unpinned', () => {
  it('should throw when path is missing', () => {
    expect(() => unpinned()).toThrow()
  })

  it('should throw if package.json is not at given path', () => {
    expect(() => unpinned('./package.json')).toThrow()
  })

  it('should return object with unpinned dependencies', async () => {
    const path = resolve(__dirname, '__fixtures__/package.failure.json')
    await expect(unpinned(path)).toMatchSnapshot()
  })

  it('should return undefined if all dependencies are pinned', async () => {
    const path = resolve(__dirname, '__fixtures__/package.success.json')
    await expect(unpinned(path)).toMatchSnapshot()
  })
})
