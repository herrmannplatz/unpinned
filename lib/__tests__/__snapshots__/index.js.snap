// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`unpinned should return object with unpinned dependencies 1`] = `
Object {
  "dependencies": Object {
    "moduleA": "^1.0.0",
  },
  "devDependencies": Object {
    "moduleB": "~1.0.0",
  },
  "optionalDependencies": Object {
    "moduleD": "*",
  },
  "peerDependencies": Object {
    "moduleC": "1.x.x",
  },
}
`;

exports[`unpinned should return undefined if all dependencies are pinned 1`] = `undefined`;
